compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -465,6 +465,40 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             );
         }
 
+        // Collect all possible candidates in the chain with same 'item_name'
+        let mut all_methods: Vec<super::probe::Candidate<'_>> = vec![];
+        if let Mode::MethodCall = mode && let SelfSource::MethodCall(mut rcvr_expr) = source {
+        while let hir::ExprKind::MethodCall(_path_segment, parent_expr, _args, _span) =
+        rcvr_expr.kind
+        {
+            rcvr_expr = parent_expr;
+            let prev_ty = self.resolve_vars_if_possible(
+                self.typeck_results
+                    .borrow()
+                    .expr_ty_adjusted_opt(rcvr_expr)
+                    .unwrap_or(Ty::new_misc_error(self.tcx)),
+            );
+
+             all_methods.extend(self.probe_for_name_many(
+                Mode::MethodCall,
+                item_name,
+                None,
+                IsSuggestion(true),
+                prev_ty,
+                rcvr_expr.hir_id,
+                ProbeScope::AllTraits,));
+        }
+        // Check if any of the candidates match
+        // 1. If one or more match both the parameters and return type, emit a span_note at the level of the candidate of a possible match
+        // 2. If none match per case 1. but there are name matches, (ie., the 'item_name' matches one or more candidates), emit a span note with a suggestion
+
+        for matched_method in &all_methods {
+            //err.span_note(sp: matched_method, msg:format!("Possible match: {matched_method:?}"));
+            println!("At the span of the matched_method: {matched_method:?}");
```

Comment:
`tcx.def_span(matched_method.item.def_id)` will give you the span of where the method is defined and `tcx.def_path_str(matched_method.item.def_id)` will give you a `String` with the `path::to::Type::method` that you would want to present to the user in the message.

We already know that the `Self` type is correct (`probe_for_name` will take care of that). The only thing left might be to try and do some check with `self.can_eq(type_after_call, matched_method.xform_ret_ty.unwrap())`, but we can likely leave that for a later PR.